controllers/cluster_controller.go

```diff
@@ -89,6 +97,48 @@ func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 func (r *ClusterReconciler) reconcile(ctx context.Context, cluster *capiv1alpha3.Cluster) (ctrl.Result, error) {
 	r.Log.Debugf(ctx, "calling reconcile")
+
+	migrator, err := r.MigratorFactory.NewMigrator(*cluster)
+	if err != nil {
+		return ctrl.Result{}, microerror.Mask(err)
+	}
+
+	alreadyMigrated, err := migrator.IsMigrated(ctx)
+	if err != nil {
+		return ctrl.Result{}, microerror.Mask(err)
+	}
+
+	if alreadyMigrated {
+		r.Log.Debugf(ctx, "cluster is already migrated")
```

Comment:
This log feels pretty random when other `if`s don't emit any logs IMO.

Reply:
added more logs